packages/account-web-vault/src/web-user.ts

```diff
@@ -0,0 +1,250 @@
+import { Network, Context, Profile } from '@verida/client-ts';
+import { VaultAccount, hasSession, VaultAccountConfig } from './index';
+import { EventEmitter } from 'events'
+import { MessageSendConfig } from '@verida/client-ts/dist/context/interfaces';
+
+const VeridaEvents = new EventEmitter()
+
+export interface WebUserProfile {
+    name?: string
+    avatarUri: string
+    country?: string
+    description?: string
+}
+
+export interface WebUserConfig {
+    accountConfig: VaultAccountConfig
+    clientConfig: ClientConfig
+    contextConfig: ContextConfig
+    networkConfig: NetworkConnectionConfig
+}
+
+export interface WebUserMessage {
+    subject: string
+    text: string
+    link?: WebUserMessageLink
+}
+
+export interface WebUserMessageLink {
+    url: string
+    text: string
+}
+
+/**
+ * 
+ * @event profileChanged
+ * @event connected
+ * @event disconnected
+ */
+export class WebUser {
+
+    private static config: WebUserConfig
+    private static context?: Context
+    private static account?: VaultAccount
+    private static profile?: WebUserProfile
+    private static did?: string
+
+    private static connecting: Promise<boolean>
+    private static profileConnection: Profile
+
+    public static configure(config: WebUserConfig) {
+        WebUser.config = config
+    }
+
+    public static async getContext() {
+        await WebUser.requireConnection()
+        return WebUser.context
```

Comment:
If we are absolutely sure that `context` is defined when we are connected, then we should narrow the type of context to `Context` instead of `Context | undefined`.

This would greatly help the DX not having to check if `context` is undefined all the time even though the account is connected and the context is there.

Reply:
Good point.